src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -849,5 +851,23 @@ internal static bool ConsideringTupleNamesCreatesDifference(Symbol member1, Symb
             return !CSharpWithTupleNamesComparer.Equals(member1, member2) &&
                 CSharpWithoutTupleNamesComparer.Equals(member1, member2);
         }
+
+        private enum RefKindMatching
```

Comment:
>RefKindMatching [](http://example.com/codeflow?start=21&length=15)

The names look confusing to me. I would call the enum "RefKindCompareMode", or something similar. Made it Flags, with members: "DoNotConsiderDifferences - 0", "ConsiderDifferences - 1", "AllowRefReadonlyVsInMismatch - 2"